SU2_CFD/include/fluid/CFluidModel.hpp

```diff
@@ -119,10 +125,39 @@ class CFluidModel {
    */
   su2double GetCv() const { return Cv; }
 
+  /*!
+   * \brief Get the source term of the transported scalar
+   */
+  virtual inline su2double GetSourceScalar(int){ return 37; }
+  
+  /*!
+  * \brief Get the number of transported scalars
+  */
+  virtual inline unsigned short GetNScalars() {return 73; }
+
+  /*!
+   * \brief Get the looked up scalar field for combustion
+   */
+  virtual inline su2double GetLookupScalar(int){ return 37; }
+
+
+  virtual CLookUpTable* GetLookUpTable() {return look_up_table; }
+
+  virtual inline unsigned long GetEnthFromTemp(su2double *enthalpy, 
+                                               su2double  val_prog, 
+                                               su2double  val_temp) { return 73; }
+
+  virtual inline pair<su2double, su2double> GetTableLimitsEnth() { return make_pair(73,37); }
+  virtual inline pair<su2double, su2double> GetTableLimitsProg() { return make_pair(73,37); }
+
+  virtual inline su2double GetdDensitydPV() { return 0.0; }
+  virtual inline su2double GetdSourcePVdPV() { return 0.0; }
+  virtual inline su2double GetdDensitydEnth() { return 0.0; }
+
   /*!
    * \brief Get fluid dynamic viscosity.
    */
-  su2double GetLaminarViscosity() {
+  virtual inline su2double GetLaminarViscosity() {
     LaminarViscosity->SetViscosity(Temperature, Density);
     Mu = LaminarViscosity->GetViscosity();
```

Comment:
I do not like this architecture because now it is double polymorphic.
i.e. the method is virtual therefore cannot be inlined, and then it makes calls to the models which are again polymorphic.

Is it possible to avoid this by creating ViscosityModels and so on that are backed by the lookup table?